app/models/registration_payment.rb

```diff
@@ -11,23 +11,27 @@ class RegistrationPayment < ApplicationRecord
 
   delegate :auto_accept_preference_live?, to: :registration
   after_create :auto_accept_hook, if: :auto_accept_preference_live?
-  after_create :auto_close_hook, unless: :refunded_registration_payment_id?
+  after_save :auto_close_hook
```

Comment:
And as an afterthought to my above comment, what about an `after_create ..., if: :is_completed?` as well as a separate `after_update ..., if: :saved_change_to_is_completed?`